packages/react-client/src/ReactFlightClient.js

```diff
@@ -623,6 +625,23 @@ function createElement(
       writable: true,
       value: null,
     });
+    if (enableOwnerStacks) {
+      Object.defineProperty(element, '_debugStack', {
+        configurable: false,
+        enumerable: false,
+        writable: true,
+        value: {stack: stack},
+      });
+      Object.defineProperty(element, '_debugTask', {
+        configurable: false,
+        enumerable: false,
+        writable: true,
+        value: null,
+      });
+    }
+    // TODO: We should be freezing the element but currently, we might write into
+    // _debugInfo later. We could move it into _store which remains mutable.
+    Object.freeze(element.props);
```

Comment:
On a Next.js client side navigations, we get the following RSC payload:
```
3:I["(app-pages-browser)/../../../../packages/next/dist/client/components/client-page.js",["app-pages-internals","static/chunks/app-pages-internals.js"],"ClientPageRoot"]
4:I["(app-pages-browser)/./app/[id]/page.tsx",["app/[id]/page","static/chunks/app/%5Bid%5D/page.js"],"default",1]
5:I["(app-pages-browser)/../../../../packages/next/dist/client/components/layout-router.js",["app-pages-internals","static/chunks/app-pages-internals.js"],""]
6:I["(app-pages-browser)/../../../../packages/next/dist/client/components/render-from-template-context.js",["app-pages-internals","static/chunks/app-pages-internals.js"],""]
2:{"name":"","env":"Server","owner":null}
1:D"$2"
8:{"name":"","env":"Server","owner":null}
7:D"$8"
0:["development",[["children",["id","a","d"],[["id","a","d"],{"children":["__PAGE__",{}]}],[["id","a","d"],{"children":["__PAGE__",{},[["$L1",["$","$L3",null,{"props":{"params":{"id":"a"},"searchParams":{}},"Component":"$4"},null]],null],null]},["$","$L5",null,{"parallelRouterKey":"children","segmentPath":["children","$0:1:0:3:0","children"],"error":"$undefined","errorStyles":"$undefined","errorScripts":"$undefined","template":["$","$L6",null,{},null],"templateStyles":"$undefined","templateScripts":"$undefined","notFound":"$undefined","notFoundStyles":"$undefined","styles":null},null],null],[null,"$L7"]]]]
9:"$Sreact.fragment"
7:["$","$9","19VxPb-E-LbHeAfz9ZCJ2",{"children":[["$","meta","0",{"name":"viewport","content":"width=device-width, initial-scale=1"},null],["$","meta","1",{"charSet":"utf-8"},null]]}]
1:null
```

Which then results in ""Cannot assign to read only property 'Component' of object '#<Object>'" thrown from https://github.com/facebook/react/blob/6c409acefde29d6ef87dfb208716ec3272bd3c54/packages/react-client/src/ReactFlightClient.js#L644

I don't have a smaller repro yet. Is this a bug with the Flight Server, our usage or should we just not freeze the props on the Flight Client?

Reply:
I think I'm seeing a similar here https://github.com/hi-ogawa/vite-plugins/pull/318 I don't have a repro outside of my framework yet, but the error seems related to passing client reference as client component props like below:

- `page.tsx`

```ts
import { Client1, Client2 } from "./_client";

export default function Page() {
  return (
    <div>
      <h4 className="font-bold">Repro</h4>
      {/* TypeError: Cannot assign to read only property 'SomeComp' of object '#<Object>' */}
      <Client1 SomeComp={Client2} />
    </div>
  );
}
```

- `_client.tsx`

```ts
"use client";

import React from "react";

export function Client1(props: { SomeComp: React.ComponentType }) {
  return (
    <div>
      [Client1] props.SomeComp = <props.SomeComp />
    </div>
  );
}

export function Client2() {
  return <span>[Client2]</span>;
}
```

I'm using a similar pattern for Error boundary and flight client seems to be throwing for this.